lib/membrane/video_compositor/handler.ex

```diff
@@ -59,6 +61,16 @@ defmodule Membrane.VideoCompositor.Handler do
               state :: state()
             ) :: immediate_callback_return() | idle_callback_return()
 
+  @doc """
+  Callback invoked upon expiration of Temporal Scene.
+
+  See `Membrane.VideoCompositor.Scene.Temporal`.
+  """
+  @callback handle_scene_expire(
+              ctx :: CallbackContext.SceneExpire.t(),
+              state :: state()
+            ) :: immediate_callback_return()
+
```

Comment:
This may be inconsistent with our callbacks API. In other callbacks, we pass an event related to that callback invocation as a first argument, while here we don't. One may argue, that the `ctx` contains an expired scene, and passing it once more is repetitive and unnecessary, but I think we might solve it. The `ctx` contains
```elixir
current_scene: Membrane.VideoCompositor.Scene.t()
```
meaning we can't pass there
```elixir
Membrane.VideoCompositor.Scene.Temporal.t()
```

I imagine, that sometimes it might be helpful in this callback to get access not only to the last scene but also to the whole expired scene specification (e.g. to the expired `sequence`). Therefore, I would opt for that API:
```suggestion
  @doc """
  Callback invoked upon expiration of Temporal Scene.

  See `Membrane.VideoCompositor.Scene.Temporal`.
  """
  @callback handle_scene_expire(
              expired_scene :: Membrane.VideoCompositor.Scene.Temporal.t(),
              ctx :: CallbackContext.SceneExpire.t(),
              state :: state()
            ) :: immediate_callback_return()

```

But here we run into some ambiguity, that I discovered while writing #112. Typing `CallbackContext.SceneExpire.t()` might suggest, that it holds expired `Membrane.VideoCompositor.Scene.Temporal.t()`, while it doesn't. This is quite problematic since all our callbacks receive `Membrane.VideoCompositor.Handler.CallbackContext.[callback_name].t()`. As we see here and in #112 , this certainly isn't the perfect solution, with the possibility of being misleading.

I suggest that we change our callback API, to the one proposed in #114 .  If implemented in this PR, we should also modify the PR name, since it's certainly outside of the scope of `designing the temporal scene API`.